pkg/apis/pravega/v1alpha1/pravegacluster_types.go

```diff
@@ -103,6 +107,10 @@ type ClusterSpec struct {
 	//pravega-bookie-2.pravega-bookie-headless.default:3181
 	BookkeeperUri string `json:"bookkeeperUri"`
 
+	// SupportedUpgradePaths is the name of the configMap which contains the list of
+	// supported upgrade paths for all versions of the pravega cluster
+	SupportedUpgradePaths string `json:"supportedUpgradePaths"`
```

Comment:
I don't see the need for this variable or its default value. We can have a **const** defined for the config-map name and use that. In any Pravega installation user would be required to create only **1 config map** for checking supported upgrade paths and we can mandate that this config map must be created with this name. Please choose a suitable name like "pravega-supported-upgrade-paths".  We need to explain this in the upgrade documentation as well.